meta-mender-core/recipes-bsp/grub-mender-grubenv/grub-mender-grubenv.inc

```diff
@@ -93,3 +94,19 @@ do_install() {
 
     oe_runmake -f ${S}/Makefile srcdir=${S} ENV_DIR=${GRUB_CONF_LOCATION} DESTDIR=${D} install
 }
+
+addtask sign after do_install before do_deploy do_package
+python do_sign() {
+    if bb.utils.contains('DISTRO_FEATURES', 'efi-secure-boot', True, False, d):
+        uks_bl_sign("%s%s/grub.cfg" % (d.getVar("D"), d.getVar("GRUB_CONF_LOCATION")), d)
+}
+do_sign[prefuncs] += "${@bb.utils.contains('DISTRO_FEATURES', 'efi-secure-boot', 'check_deploy_keys', '', d)}"
+do_sign[prefuncs] += "${@'check_boot_public_key' if d.getVar('GRUB_SIGN_VERIFY', True) == '1' else ''}"
+
+fakeroot do_chownboot() {
+    if "${@bb.utils.contains('DISTRO_FEATURES', 'efi-secure-boot', 'true', 'false', d)}"; then
+        chown root:root -R "${D}${GRUB_CONF_LOCATION}/grub.cfg${SB_FILE_EXT}"
+    fi
+}
+addtask chownboot after do_sign before do_package
```

Comment:
I believe you can simplify this considerably by doing this instead:

```python
python() {
    if bb.utils.contains('DISTRO_FEATURES', 'efi-secure-boot', True, False, d):
        bb.build.addtask('do_sign', 'do_deploy do_package', 'do_install', d)
        bb.build.addtask('do_chownboot', 'do_package', 'do_sign', d)
}
```

This conditional adding means that you can get rid of all the `efi-secure-boot` checks elsewhere.

Reply:
Fixed in the latest push.